### ISSUE TITLE
undefined join type AssertionFailure

### ISSUE THREAD

**Issue by robsosno** [nhibernate/nhibernate-core#3294]
After simplification I have following LINQ query:
```
        var q = from c in Orm.DB.Cards
                from s in Orm.DB.Services
                where c.Id == 1 && s.RowSerno == c.Id
                select s;
```
Entities Cards and Services are unrelated, i.e. there is no Many-to-one or bag on either side. s.Rowserno is not a primary key of Services entity neither is part of it. c.Id is primary key.
Previously it worked without problems. After upgrade I have "undefined join type" error.
Stack trace attached: 
[err1.txt](https://github.com/nhibernate/nhibernate-core/files/11334902/err1.txt)

What does it mean "upgrade":
Previously I had .NET Core 3.1, NHibernate 5.2.5.0, database Informix, driver IBM.Data.DB2.Core 3.1.0.500
Now I have .NET 6.0, NH 5.4.1.0, same database, driver Net.IBM.Data.Db2 6.0.0.300
 
Similar problem is with JoinEntityAlias. I'll create separate issue as I'm not sure if this is the same problem or not.

**Comment by bahusoid** [comment]
What dialect is used? Try with [`InformixDialect1000`](https://github.com/nhibernate/nhibernate-core/blob/master/src/NHibernate/Dialect/InformixDialect1000.cs)

**Comment by robsosno** [comment]
Good point, I haven't noticed that I have NHibernate.Dialect.InformixDialect . Of course it should be InformixDialect1000.
Tomorrow I'll check this. Thanks!

**Comment by bahusoid** [comment]
Fixed by #3313
